ModularTegustation/tegu_mobs/lc13_outskirtdwellers.dm

```diff
@@ -447,3 +447,598 @@
+	var/attack_delay = 0.5 SECONDS //0.5 seconds at stage 1, 1 second at stage 2
+	var/countering = FALSE //are we
+
+	var/counter_threshold = 500 //300 at stage 2
+	var/counter_ready = FALSE //are we ready to counter?
+
+	var/counter_speed = -2 //subtracted from the movedelay when dashing
+	var/original_speed = 6 //used to reset speed after dashing
+
+	var/lovewhip_damage = 100
+	var/damage_taken
+
+/mob/living/simple_animal/hostile/lovetown/abomination/proc/StageTransition()
+	icon_living = "lovetown_abomination2"
+	if(!countering && can_act)
+		icon_state = icon_living
+	current_stage = 2
+	move_to_delay = 4
+	counter_speed = -1
+	original_speed = 4
+	attack_delay = 1 SECONDS
+	counter_threshold = 300
+	playsound(get_turf(src), 'sound/creatures/lc13/lovetown/abomination_stagetransition.ogg', 75, 0, 3)
+
+/mob/living/simple_animal/hostile/lovetown/abomination/proc/AoeAttack() //all attacks are an AoE when not dashing
+	can_act = FALSE
+	face_atom(target)
+	playsound(get_turf(src), 'sound/abnormalities/apocalypse/swing.ogg', 75, 0, 3)
+	switch(current_stage)
+		if(1)
+			icon_state = "lovetown_abomination_slamwindup"
+		if(2)
+			icon_state = "lovetown_abomination_slamwindup2"
+	SLEEP_CHECK_DEATH(attack_delay) //takes longer to slam on phase 2
+	for(var/turf/T in view(current_stage, src))//scales with stage, at stage 2 hits 2 tiles around
+		new /obj/effect/temp_visual/lovetown_shapes(T)
+		HurtInTurf(T, list(), (rand(melee_damage_lower, melee_damage_upper)/2), RED_DAMAGE, check_faction = TRUE, hurt_mechs = TRUE) //30~40 damage
+	switch(current_stage)
+		if(1)
+			icon_state = "lovetown_abomination_slam"
+		if(2)
+			icon_state = "lovetown_abomination_slam2"
+	playsound(get_turf(src), 'sound/abnormalities/mountain/slam.ogg', 75, 0, 3)
+	SLEEP_CHECK_DEATH(0.4 SECONDS)
+	icon_state = icon_living
+	can_act = TRUE
+
+/mob/living/simple_animal/hostile/lovetown/abomination/proc/DashCounter() //increases move speed and hits with a powerful attack that knocks back far away
+	playsound(get_turf(src), 'sound/creatures/lc13/lovetown/abomination_counter_start.ogg', 75, 0, 3)
+	switch(current_stage)
+		if(1)
+			icon_state = "lovetown_abomination_dash"
+		if(2)
+			icon_state = "lovetown_abomination_dash2"
+	countering = TRUE
+	counter_ready = FALSE
+	move_to_delay += counter_speed
+	visible_message("<span class='warning'>[src] sprints toward [target]!</span>", "<span class='notice'>You quickly dash!</span>", "<span class='notice'>You hear heavy footsteps speed up.</span>")
+	addtimer(CALLBACK(src, .proc/DisableCounter, original_speed), 4 SECONDS) //disables the counter after 4 seconds
+
+/mob/living/simple_animal/hostile/lovetown/abomination/proc/DisableCounter(original_speed) //resets the speed
+	if(countering)
+		move_to_delay = original_speed
+		countering = FALSE
+		playsound(get_turf(src), 'sound/creatures/lc13/lovetown/abomination_counter_end.ogg', 75, 0, 3)
+		SLEEP_CHECK_DEATH(10)
+		icon_state = icon_living
```

Comment:
```suggestion
	addtimer(CALLBACK(src, .proc/DisableCounter), 4 SECONDS) //disables the counter after 4 seconds

/mob/living/simple_animal/hostile/lovetown/abomination/proc/DisableCounter() //resets the speed
	if(countering)
		move_to_delay = original_speed
		countering = FALSE
		playsound(get_turf(src), 'sound/creatures/lc13/lovetown/abomination_counter_end.ogg', 75, 0, 3)
		SLEEP_CHECK_DEATH(10)
		icon_state = icon_living
```
Proc argument is redundant because the variable is assigned to the mob already so it's already accessible

Reply:
Done.